File: src/basic/Pyhthon3/bfs.py
visited = dict(zip([v for v in G], [False for _ in G])) # 방문 여부를 체크하는 배열

def bfs(v): # 정점 v부터 탐색 시작
  visited[v] = True
  queue = [v]
  while queue:
    now = queue.pop(0)
    for i in G[now]: # 현재 선택한 정점의 모든 자식 정점을 방문.
      if not visited[i]:
        visited[i] = True
        queue.append(i)

File: src/test/BFS.java
package test;

import java.util.Iterator;
import java.util.LinkedList;

public class BFS {
	class Graph {
		private int V;
		private LinkedList<Integer> adj[];
		
		Graph(int v) {
			V = v;
			adj = new LinkedList[v];
			for (int i = 0; i < v; ++i)
				adj[i] = new LinkedList();
		}
		
		void addEdge(int v, int w) { adj[v].add(w); }
		
		/* BFS */
		void BFS(int s) {
			boolean visited[] = new boolean[V];
			LinkedList<Integer> queue = new LinkedList<Integer>();
			
			visited[s] = true;
			queue.add(s);
			
			while (queue.size() != 0) {
				// 방문한 노드를 큐에서 추출(dequeue)하고 값을 출력
				s = queue.poll();
				System.out.print(s + " ");
				
				// 방문한 노드와 인접한 모든 노드를 가져온다.
				Iterator<Integer> i = adj[s].listIterator();
				while (i.hasNext()) {
					int n = i.next();
					// 방문하지 않은 노드면 방문한 것으로 표시하고 큐에 삽입(enqueue)
					if (!visited[n]) {
						visited[n] = true;
						queue.add(n);
					}
				}
			}
		}
	}
}
